### PR TITLE
Add support for Branded Fares Upsell API

### LINKED ISSUE #100: Add support for Branded Fares Upsell API
https://developers.amadeus.com/self-service/category/air/api-doc/branded-fares-upsell/api-reference

### PR DESCRIPTION
**Pull request by xianqiliu**
Fixes #100 
This PR is ready for review, please go ahead and thank you!

### CODE REVIEW

File: src/main/java/com/amadeus/shopping/flightOffers/Upselling.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import com.amadeus.Response;
+import com.amadeus.exceptions.ResponseException;
+import com.amadeus.resources.FlightOfferSearch;
+import com.amadeus.resources.Resource;
+import com.google.gson.JsonObject;
+
+/**
+ * <p>
+ * A namespaced client for the
+ * <code>/v1/shopping/availability/flight-availabilities</code> endpoints.
```
Comment (tsolakoua) [inline src/main/java/com/amadeus/shopping/flightOffers/Upselling.java:13]: Typo from other API

File: src/main/java/com/amadeus/resources/FlightOfferSearch.java [hunk 2]
```diff
@@ -16,8 +16,10 @@ protected FlightOfferSearch() {}
   private @Getter String id;
   private @Getter String source;
   private @Getter boolean instantTicketingRequired;
+  private @Getter boolean disablePricing;
```
Comment (tsolakoua) [inline src/main/java/com/amadeus/resources/FlightOfferSearch.java:19]: Screenshot 2022-03-23 at 17 41 54
Why don't you create the new resource for the API?

Comment (xianqiliu) [reply src/main/java/com/amadeus/resources/FlightOfferSearch.java:19]: Hello Anna! From my understanding, the picture you show above is the input body for this API.

Comment (xianqiliu) [reply src/main/java/com/amadeus/resources/FlightOfferSearch.java:19]: @tsolakoua Hello! From my understanding, I think the picture you commented on above is the input body for this API.

Comment (xianqiliu) [reply src/main/java/com/amadeus/resources/FlightOfferSearch.java:19]: After testing on the postman, I think the result for this API is actually matching the Flight Offer Search Model.
Besides, you might be confused why I add lots of new properties into this Model. The reason is that actually the model is supposed to contain these new properties cause the original model from [Flight Offer Search API ](https://developers.amadeus.com/self-service/category/air/api-doc/flight-offers-search/api-reference) also contains those properties.
image

Comment (xianqiliu) [reply src/main/java/com/amadeus/resources/FlightOfferSearch.java:19]: And you can also find the data model is actually the Flight-Offer in the result of [Branded fares upsell API](https://developers.amadeus.com/self-service/category/air/api-doc/branded-fares-upsell/api-reference).
image

**Comment by xianqiliu** [comment]
And I also merged the new code from the original repo into this PR. 
Please review and thank you!

**Review by tsolakoua — APPROVED** [review]

**Pull request merged**